### PR TITLE
Remove remaining `malloc` calls and globally enable MARK_SWEEP

### PR DESCRIPTION
**Pull request by scallyw4g**

### CODE REVIEW

**Comment by scallyw4g** [comment]
Looks like everything's good to me.

Next step is incrementally turning on ASAN failures if we leak in tests

File: mycpp/marksweep_gc_test.cc [hunk 1]
```diff
@@ -35,6 +35,35 @@ TEST string_collection_test() {
...
+    Str *test_str0 = nullptr;
+    Str *test_str1 = nullptr;
+    StackRoots _roots({&test_str0, &test_str1});
+
+    test_str0 = StrFromC("foo");
+    test_str1 = StrFromC("foo");
+
+    for (int i = 0; i < KiB(32); ++i)
+    {
+      if (i % 2)
```
Comment (andychu) [inline mycpp/marksweep_gc_test.cc:48]: What is the purpose of this test?  It could use some comments

File: mycpp/TEST.sh [hunk 2]
```diff
@@ -240,11 +240,6 @@ compare-examples() {
 }
 
 test-sweep-asan-leaks() {
-  ninja _bin/cxx-sweepasan/mycpp/examples/fib_iter.mycpp
-  ASAN_OPTIONS='' _bin/cxx-sweepasan/mycpp/examples/fib_iter.mycpp
-
-  # CURRENTLY 9 of 22 fail
-  # This does NOT check if all pass
   ASAN_OPTIONS='' examples-variant '' sweepasan
```
Comment (andychu) [inline mycpp/TEST.sh:243]: How many of these pass with ASAN now?  All?  

I think this comment is saying that it doesn't exit 1 if there are failures, which we should fix

So we shouldn't delete the comment yet

**Comment by andychu** [comment]
This mostly looks good, we can sync up about the rooting and how to enable ASAN leak detection gradually

File: mycpp/marksweep_gc_test.cc (hunk 1 above)
Comment (scallyw4g) [reply mycpp/marksweep_gc_test.cc:48]: Oh it was just some random junk I typed in to try and find bugs.  I'll delete it, it's not testing anything specific

File: mycpp/TEST.sh (hunk 2 above)
Comment (scallyw4g) [reply mycpp/TEST.sh:243]: There's still one failing test, but it's got nothing to do with the GC.  There's some pretty questionable code in the pretty-printers which leak and is not super easy to fix.

**Pull request merged**